tools/deploy_firmware.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/env python3
+
+import time
+import sys
+import os
+import subprocess
+import difflib
+import argparse
+import shutil
+from typing import List
+
+import pexpect
+import pexpect.fdpexpect
+import serial
+
+
+class PrplwrtDevice:
+    """Represents a prplWrt device.
+
+        Offers methods to check if a device needs to be upgraded and to do the actual upgrade.
+
+        It needs to have access to the artifacts of a build job to determine when an upgrade
+        is needed (see `artifacts_dir`).
+    """
+
+    shell_prompt = "_pexpect_prompt_ "
+    """We will temporarily use this prompt on the device to make it easier to "expect" it."""
+
+    baudrate = 115200
+    """The baudrate of the serial connection to the device."""
+
+    def __init__(self, device: str, name: str, image: str, username: str = "root"):
+        """
+
+        Parameters
+        -----------
+        device: str
+            The name of the platform (example: netgear-rax40).
+        name: str
+            The name of the device (it should ne reachable through ssh without a password).
+        image: str
+            The name of the image that can be used to upgrade the device.
+        username: str, optional
+            The username to use when connecting to the device over SSH.
+        """
+        self.device = device
+        self.name = name
+        self.image = image
+        self.username = username
+
+        self.scriptdir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
+        self.rootdir = os.path.realpath(os.path.join(self.scriptdir))
```

Comment:
Something fishy here...
```suggestion
        self.rootdir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
```

Reply:
Indeed, thanks.